File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"
#include <string>

namespace raindrops {
    std::string convert(int number) {
        std::string converted = "";
        if (number % 3 == 0)
            converted += "Pling";
        
        if (number % 5 == 0)
            converted += "Plang";
        
        if (number % 7 == 0)
            converted += "Plong";

        return (converted.size() > 0 ? converted : std::to_string(number));
    }
}  // namespace raindrops

File: solutions/zig/raindrops/1/raindrops.zig
const std = @import("std");

pub fn convert(buffer: []u8, comptime n: u32) []const u8 {
    const pling = if (n % 3 == 0) "Pling" else "";
    const plang = if (n % 5 == 0) "Plang" else "";
    const plong = if (n % 7 == 0) "Plong" else "";
    const message = pling ++ plang ++ plong;
    
    return if (message.len > 0) message else std.fmt.bufPrint(buffer, "{}", .{n}) catch unreachable;
}
